src/trackinfoobject.h

```diff
@@ -307,6 +325,14 @@ class TrackInfoObject : public QObject {
     // TrackDAO
     void setId(TrackId id);
 
+    // Stores a pointer to the original track for temporary instances that
+    // have been created by cloning another track with cloneTemporary().
+    // Ensures that the lifetime of cloned instances does not exceed the
+    // lifetime of their parent tracks. Especially important for parent
+    // tracks that are managed by TrackCache, because tags might be written
+    // back into the file when the track is evicted from the cache.
+    const TrackPointer m_pParentTrack;
```

Comment:
This would violate the rule that TrackCache keeps track of all track 
references to decide when it is safe to write tags back into the file.

Of course, if you use newTemporary() instead of cloneTemporary() you are 
still able to access a file completely disconnected from TrackCache. Maybe 
we should define that temporary track objects do not provide any guarantees 
regarding safe file access.

On 02/21/2016 02:53 PM, Daniel Schürmann wrote:
>
> In src/trackinfoobject.h 
> <https://github.com/mixxxdj/mixxx/pull/898#discussion_r53566743>:
>
> > @@ -307,6 +325,14 @@ class TrackInfoObject : public QObject {
> >      // TrackDAO
> >      void setId(TrackId id);
> >
> > +    // Stores a pointer to the original track for temporary instances that
> > +    // have been created by cloning another track with cloneTemporary().
> > +    // Ensures that the lifetime of cloned instances does not exceed the
> > +    // lifetime of their parent tracks. Especially important for parent
> > +    // tracks that are managed by TrackCache, because tags might be written
> > +    // back into the file when the track is evicted from the cache.
> > +    const TrackPointer m_pParentTrack;
>
> I have still not understand why this is required and why we have a problem.
> When the parent track falls out of scope and is deleted, it was not needed 
> any more. Why is it require to keep it alive?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/898/files#r53566743>.
>